packages/bpk-component-card-list/src/BpkCardList.tsx

```diff
@@ -0,0 +1,170 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ReactElement } from 'react';
+import { useState } from 'react';
+
+import BpkBreakpoint, { BREAKPOINTS } from '../../bpk-component-breakpoint';
+import { BpkButtonV2 } from '../../bpk-component-button';
+import BpkSectionHeader from '../../bpk-component-section-header';
+import { cssModules } from '../../bpk-react-utils';
+
+import BpkCardListGrid from './BpkCardListGrid';
+import BpkCardListRail from './BpkCardListRail';
+import BpkCardListRow from './BpkCardListRow';
+import BpkCardListStack from './BpkCardListStack';
+import { BpkAccessoryTypes, type BpkCardListProps } from './common-types';
+
+import STYLES from './BpkCardList.module.scss';
+
+const getClassName = cssModules(STYLES);
+const MAX_ITEMS = 12; // MAX should be 12 for Desktop Grid and Mobile Stack
```

Comment:
Restricting the number of cards that can be displayed feels like we're unnecessarily reducing the flexibility and harming the re-useability of this component.

 How did we choose that the maximum is 12, and why is there a limit in the first place?

Should it instead be the decision of the consumer of the component to decide how many the maximum number of cards is appropriate based on the context of the where the card list is being used?

Reply:
This limit has come from design as part of the setup and specific interaction pattern designed, so if you feel this is restrictive, I would bring the conversation to designers so they can figure out to remove the limit or keep it

![Screenshot 2024-06-14 at 12 33 40](https://github.com/Skyscanner/backpack/assets/8831547/7e87c428-e590-4262-801a-fedb301ef9de)